File: app/src/main/java/com/mm/mm/MyAdapter.java
package com.mm.mm;

import android.view.*;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

/**
 * MyAdapter — Adapter RecyclerView.
 * TODO: Replace 'String' with your actual data model class
 */
public class MyAdapter extends RecyclerView.Adapter<MyAdapter.ViewHolder> {

    public interface OnItemClickListener {
        void onItemClick(String item, int position);
    }

    private final List<String> dataList;
    private OnItemClickListener listener;

    public MyAdapter(List<String> dataList) { this.dataList = dataList; }

    public void setOnItemClickListener(OnItemClickListener l) { this.listener = l; }

    @NonNull @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_list, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        String item = dataList.get(position);
        holder.tvTitle.setText(item);         // TODO: customize
        holder.tvSubtitle.setText("Item #" + (position + 1)); // TODO: customize
        holder.itemView.setOnClickListener(v -> {
            if (listener != null) listener.onItemClick(item, position);
        });
    }

    @Override
    public int getItemCount() { return dataList != null ? dataList.size() : 0; }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTitle, tvSubtitle;
        public ViewHolder(@NonNull View v) {
            super(v);
            tvTitle    = v.findViewById(R.id.tvTitle);
            tvSubtitle = v.findViewById(R.id.tvSubtitle);
        }
    }
}

File: app/src/main/java/com/tes/jk/MyAdapter.kt
package com.tes.jk

import android.view.*
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

/**
 * MyAdapter — Adapter RecyclerView.
 * TODO: Ganti String dengan model data Anda
 */
class MyAdapter(
    private val dataList: List<String>,
    private val onItemClick: (String, Int) -> Unit = { _, _ -> }
) : RecyclerView.Adapter<MyAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvTitle: TextView    = view.findViewById(R.id.tvTitle)
        val tvSubtitle: TextView = view.findViewById(R.id.tvSubtitle)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_list, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = dataList[position]
        holder.tvTitle.text    = item             // TODO: sesuaikan
        holder.tvSubtitle.text = "Item #${position + 1}" // TODO: sesuaikan
        holder.itemView.setOnClickListener { onItemClick(item, position) }
    }

    override fun getItemCount() = dataList.size
}
